### ISSUE TITLE
ChangeFeedIterator gets stuck on multihashed containers for prefix partitions keys that span multiple physical partitions.

### ISSUE THREAD

**Issue by NaluTripician** [Azure/azure-cosmos-dotnet-v3#3865]
**Describe the bug**
When reading the `ChangeFeed` for a prefix partition key that spans multiple physical partitions the `ChangeFeed` will not be able to handle the partition split and will get stuck in an infinite loop.

**Expected behavior**
Results should be returned.

**Actual behavior**
`ReadNextAsync` becomes stuck in an infinite loop.

```csharp
// Create Container Hierarchical PartitionKeys
this.containerProperties = new ContainerProperties("mycoll", new List<string> { "/State", "/City", "/ZipCode" });
this.container = await database.CreateContainerAsync(this.containerProperties);

// ...
//Container is then filled with enough items span multiple physical partitions 
// ...

//Create a Prefix PartitionKey
Cosmos.PartitionKey prefixPK = new PartitionKeyBuilder().Add("Redmound").Build();

//Create ChangeFeedIterator
FeedRange feedRange = FeedRange.FromPartitionKey(shardPartitionKey);
FeedIterator<dynamic> feedIterator = container.GetChangeFeedIterator<dynamic>(ChangeFeedStartFrom.Beginning(shardFeedRange), ChangeFeedMode.Incremental, new ChangeFeedRequestOptions() { PageSizeHint = 10 });

while (feedIterator .HasMoreResults)
{
    FeedResponse<dynamic> response = await feedIterator .ReadNextAsync(); // BUG: Code gets stuck here!
}

```

**Comment by ealsur** [comment]
Related logical investigation steps that yielded the bug:

1. User sends request with PK with less components
1. RequestInvokerHandler converts the FeedRangePK to FeedRangeEPK https://github.com/Azure/azure-cosmos-dotnet-v3/blob/master/Microsoft.Azure.Cosmos/src/Handler/RequestInvokerHandler.cs#L166-L169
1. RequestInvokerHandler finds that the FeedRangeEPK spans multiple partitions: https://github.com/Azure/azure-cosmos-dotnet-v3/blob/master/Microsoft.Azure.Cosmos/src/Handler/RequestInvokerHandler.cs#L239-L251
1. Error bubbles to https://github.com/Azure/azure-cosmos-dotnet-v3/blob/master/Microsoft.Azure.Cosmos/src/Pagination/CrossPartitionRangePageAsyncEnumerator.cs#L111
1. Then it calls into the FeedRangeProvider.GetChildRanges https://github.com/Azure/azure-cosmos-dotnet-v3/blob/master/Microsoft.Azure.Cosmos/src/Pagination/CrossPartitionRangePageAsyncEnumerator.cs#L113-L118
1. Which calls into FeedRangePartitionKey and always returns a point range, even if the PK spans multiple ranges: https://github.com/Azure/azure-cosmos-dotnet-v3/blob/master/Microsoft.Azure.Cosmos/src/FeedRange/FeedRanges/FeedRangePartitionKey.cs#L25-L37 because that method is simply designed for a normal PK that never goes outside a physical partition.

**Comment by amitdhar** [comment]
Hi, any updates on this? Thanks!

**Comment by amitdhar** [comment]
Any status updates?

**Comment by NaluTripician** [comment]
Issue is fixed with latest release of the SDK (3.35.0)
